### PR TITLE
Improve dbt errors

### PR DESCRIPTION
**Pull request by guy-har**

### CODE REVIEW

File: pkg/metastore/ms_client.go [hunk 1]
```diff
@@ -55,22 +59,30 @@ func CopyDB(ctx context.Context, fromClient, toClient Client, fromDB, toDB, toBr
...
+		return transformedLocation, nil
 	}
 	return copyDBWithTransformLocation(ctx, fromClient, toClient, fromDB, toDB, transformLocation)
 }
 
 func copyDBWithTransformLocation(ctx context.Context, fromClient, toClient Client, fromDB string, toDB string, transformLocation func(location string) (string, error)) error {
 	schema, err := fromClient.GetDatabase(ctx, fromDB)
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to get database on copy: %w", err)
```
Comment (nopcoder) [inline pkg/metastore/ms_client.go:74]: would add the fromDB to the error

File: pkg/metastore/ms_client.go [hunk 2]
```diff
@@ -55,22 +59,30 @@ func CopyDB(ctx context.Context, fromClient, toClient Client, fromDB, toDB, toBr
...
 	}
 	schema.Name = toDB
 	schema.LocationURI, err = transformLocation(schema.LocationURI)
 	if err != nil {
 		return err
 	}
-	return toClient.CreateDatabase(ctx, schema)
+	err = toClient.CreateDatabase(ctx, schema)
+	if err != nil {
+		return fmt.Errorf("failed to create database with name %s and location %s   :%w", schema.Name, schema.LocationURI, err)
```
Comment (nopcoder) [inline pkg/metastore/ms_client.go:83]: - fix the spaces
- prefer to use '%s' to quote names/values

File: pkg/metastore/ms_client.go [hunk 3]
```diff
@@ -55,22 +59,30 @@ func CopyDB(ctx context.Context, fromClient, toClient Client, fromDB, toDB, toBr
 			return "", nil
 		}
 		location = HandleDBFSLocation(location, dbfsLocation)
-		return ReplaceBranchName(location, toBranch)
+		transformedLocation, err := ReplaceBranchName(location, toBranch)
+		if err != nil {
+			return "", fmt.Errorf("failed to replace branch name with location: %s and branch: %s : %w", location, toBranch, err)
```
Comment (nopcoder) [inline pkg/metastore/ms_client.go:64]: optional:
```suggestion
			return "", fmt.Errorf("failed to replace branch name with location '%s' and branch '%s': %w", location, toBranch, err)
```

File: pkg/metastore/ms_client.go [hunk 4]
```diff
@@ -44,7 +44,11 @@ type Client interface {
 func CopyOrMerge(ctx context.Context, fromClient, toClient Client, fromDB, fromTable, toDB, toTable, toBranch, serde string, partition []string, fixSparkPlaceHolder bool, dbfsLocation string) error {
 	transformLocation := func(location string) (string, error) {
 		location = HandleDBFSLocation(location, dbfsLocation)
-		return ReplaceBranchName(location, toBranch)
+		transformedLocation, err := ReplaceBranchName(location, toBranch)
+		if err != nil {
+			return "", fmt.Errorf("failed to replace branch name with location: %s and branch: %s : %w", location, toBranch, err)
```
Comment (nopcoder) [inline pkg/metastore/ms_client.go:49]:
```suggestion
			return "", fmt.Errorf("failed to replace branch name with location '%s' and branch '%s': %w", location, toBranch, err)
```

File: cmd/lakectl/cmd/dbt.go [hunk 5]
```diff
@@ -168,7 +169,7 @@ func dbtDebug(projectRoot string) string {
 	dbtCmd.Dir = projectRoot
 	output, err := dbtCmd.Output()
 	if err != nil {
-		DieErr(err)
+		DieFmt("failed to run dbt debug with output:\n %s\n and error: %s", output, err)
```
Comment (nopcoder) [inline cmd/lakectl/cmd/dbt.go:172]:
```suggestion
		DieFmt("Failed to run dbt debug with output:\n%s\nError: %s", output, err)
```

File: cmd/lakectl/cmd/dbt.go [hunk 6]
```diff
@@ -146,7 +147,7 @@ func getDbtTables(projectRoot string) ([]model, error) {
 	dbtCmd.Dir = projectRoot
 	output, err := dbtCmd.Output()
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed to run dbt debug with output:\n %s\n and error: %s", output, err)
```
Comment (nopcoder) [inline cmd/lakectl/cmd/dbt.go:150]: not sure about this one - it looks like an error that goes to the output not something that failed - it capture the output.
is that right?

**Review by nopcoder — CHANGES_REQUESTED** [review]

File: cmd/lakectl/cmd/dbt.go (hunk 6 above)
Comment (guy-har) [reply cmd/lakectl/cmd/dbt.go:150]: For some reason, when dbt debug returns an error, the information for the error is printed to the standard outpout

File: pkg/metastore/ms_client.go (hunk 1 above)
Comment (guy-har) [reply pkg/metastore/ms_client.go]: Done

File: pkg/metastore/ms_client.go (hunk 2 above)
Comment (guy-har) [reply pkg/metastore/ms_client.go]: Done

File: pkg/metastore/ms_client.go (hunk 3 above)
Comment (guy-har) [reply pkg/metastore/ms_client.go]: Done

File: pkg/metastore/ms_client.go (hunk 4 above)
Comment (guy-har) [reply pkg/metastore/ms_client.go]: Done

File: cmd/lakectl/cmd/dbt.go (hunk 5 above)
Comment (guy-har) [reply cmd/lakectl/cmd/dbt.go]: Done

File: cmd/lakectl/cmd/dbt.go (hunk 6 above)
Comment (guy-har) [reply cmd/lakectl/cmd/dbt.go]: Changed it, now, in case of error: the output is first printed as output and the error as error

**Review by nopcoder — APPROVED** [review]
LGTM

**Pull request merged**